lapce-core/src/language.rs

```diff
@@ -65,7 +65,8 @@ impl LapceLanguage {
             "php" => LapceLanguage::Php,
             "ex" | "exs" => LapceLanguage::Elixir,
             "c" | "h" => LapceLanguage::C,
-            "cpp" | "cxx" | "cc" | "hpp" | "hxx" => LapceLanguage::Cpp,
+            "cpp" | "cxx" | "cc" | "c++" | "C" | "CPP" | "hpp" | "hxx" | "hh"
```

Comment:
I think we would want to highlight "C" (and maybe "H", which is missing?) as C files, not C++

Reply:
Are extensions ever case-sensitive, e.g. should xYz be a different `LapceLanguage` than XyZ?

Reply:
I do agree with @superlou that extensions should not be case sensitive. Looking at VSCode and vim they do not treat extensions as case sensitive. At least, not for language extensions I tested. I've gone ahead and made this change which includes the changes @bugadani suggested.

If you don't want extensions to be case sensitive then I'll do just the original suggested changes.